enhancements/microshift/low-latency-workloads-on-microshift.md

```diff
@@ -0,0 +1,612 @@
+
+* As a MicroShift administrator, I want to configure MicroShift host,
+  so that I can run low latency workloads.
+
+
+### Goals
+
+Provide relatively easy way to configure system for low latency workload running on MicroShift:
+- Prepare low latency TuneD profile for MicroShift
+- Prepare necessary CRI-O configurations
+- Allow configuration of Kubelet via MicroShift config
+- Add small systemd daemon to enable TuneD profile and (optionally) reboot the host if the kernel
+  arguments change to make them effective
+  
+
+### Non-Goals
+
+- Workload partitioning (i.e. pinning MicroShift control plane components)
+- Duplicate all capabilities of Node Tuning Operator
+
+
+## Proposal
+
+To ease configuration of the system for running low latency workloads on MicroShift following
+parts need to be put in place:
+- TuneD profile
+- CRI-O configuration + Kubernetes' RuntimeClass
+- Kubelet configuration (CPU, Memory, and Topology Managers)
+- Small systemd daemon to activate TuneD profile on boot and reboot the host if the kernel args
+  are changed.
+
+New RPM will be created that will contain tuned profile, CRI-O configs, and mentioned systemd daemon.
+We'll leverage existing know how of Performance and Scalability team expertise and look at
+Node Tuning Operator capabilities.
+
+To allow customization of supplied TuneD profile for specific system, this new profile will
+include instruction to include file with variables, which can be overridden by the user.
+
+All of this will be accompanied by step by step documentation on how to use this feature,
+tweak values for specific system, and what are the possibilities and limitations.
+
+Optionally, a new subcommand `microshift doctor low-latency` might be added to main
+MicroShift binary to provide some verification checks if system configuration is matching
+expectations according to our knowledge. It shall not configure system - only report potential problems.
+
+
+### Workflow Description
+
+Workflow consists of two parts:
+1. system and MicroShift configuration
+1. preparing Kubernetes manifests for low latency
+
+#### System and MicroShift configuration
+
+##### OSTree
+
+1. User creates an osbuild blueprint:
+   - (optional) User configures `[customizations.kernel]` in the blueprint if the values are known
+     beforehand. This could prevent from necessary reboot after applying tuned profile.
+   - User adds `kernel-rt` package to the blueprint
+   - User adds `microshift-low-latency` RPM to the blueprint
+   - User enables `microshift-tuned.service`
+   - User supplies additional configs using blueprint:
+     - /etc/tuned/microshift-low-latency-variables.conf
+     - /etc/microshift/config.yaml to configure Kubelet
+1. User builds the blueprint
+1. User deploys the commit / installs the system.
+1. System boots
+1. `microshift-tuned.service` starts (after `tuned.service`, before `microshift.service`):
+   - Saves current kernel args
+   - Applies tuned `microshift-low-latency` profile
```

Comment:
Depending on what we want to include in this new tuned profile it may not be strictly necessary given tuned does ship with so many existing profiles. However, it may be worthwhile in order to give the user a clear starting point for profile configuration and modification.

Reply:
I'm not sure what you mean.
You mean that `cpu-partitioning` is not by default installed by `tuned`? We can set it as a dependency for `microshift-low-latency`.